### PR TITLE
[Design System] Add icon customisability to `DSButton`

### PR DESCRIPTION
**Pull request by alpavanoglu**
## Description

The `DSButton` component was not supporting icons before. Now it accepts an `IconName` to follow the LyzJJa6ANKToUswWmRrZee-fi-102_186.

## Testing Steps

Go to Design System Gallery in the Jetpack App.
Components > DSButton

✅ Verify if the icon version matches the designs.

## Screenshot Reference
Simulator Screenshot - iPhone 15 - 2024-03-19 at 16 36 56

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist:
- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

File: Modules/Sources/DesignSystem/Components/DSButton.swift [hunk 1]
```diff
@@ -4,19 +4,22 @@ public struct DSButton: View {
...
     @Environment(\.colorScheme) private var colorScheme
     private let title: String
+    private let iconName: IconName?
     private let style: DSButtonStyle
     @Binding private var isLoading: Bool
     private let action: (() -> Void)
 
     public init(
         title: String,
+        iconName: IconName? = nil,
```
Comment (kean) [inline Modules/Sources/DesignSystem/Components/DSButton.swift:14]: It's be great to allow using SF symbols – I need to be able to use a a standard plus sign.

Comment (salimbraksa) [reply Modules/Sources/DesignSystem/Components/DSButton.swift:14]: @kean, what do you think about adding a plus sign icon to `IconName`, and then updating the code responsible for loading the image to either load it from `Icons.xcassets` or as an SF symbol? Having all icons in one place would help in knowing which icons are used in the app.

**Review by salimbraksa — APPROVED** [review]
@alpavanoglu, the code looks good to me. However, I'm curious about whether the icon should scale when fonts are enlarged. It seems that the default iOS components do scale their icons.

File: Modules/Sources/DesignSystem/Components/DSButton.swift (hunk 1 above)
Comment (alpavanoglu) [reply Modules/Sources/DesignSystem/Components/DSButton.swift:14]: I agree with Salim. This way we can have an exhausting list whereas with direct strings, we can't know without digging into the code, which icons are being used in the app. To address the issue Salim mentioned, I'll try to come up with a solution to make the difference between a custom image and sfsymbol explicit.

**Comment by alpavanoglu** [comment]
After discussing with Salim, I've decided to take another approach to resolve the dynamic sizing for custom icons. I will follow up with another PR.

**Pull request merged**